MG-2193 - Generate mocks for proto interface

JeffMboya opened the pull request:
<!-- Copyright (c) Abstract Machines
SPDX-License-Identifier: Apache-2.0 -->

<!--
Pull request title should be `MG-XXX - description` or `NOISSUE - description` where XXX is ID of the issue that this PR relates to.
Please review the [CONTRIBUTING.md](https://github.com/absmach/magistrala/blob/main/CONTRIBUTING.md) file for detailed contributing guidelines.

For Work In Progress Pull Requests, please use the Draft PR feature, see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

For a timely review/response, please avoid force-pushing additional commits if your PR already received reviews or comments.

- Provide tests for your changes.
- Use descriptive commit messages.
- Comment your code where appropriate.
- Squash your commits
- Update any related documentation.
-->

# What type of PR is this?

This is a feature

## What does this do?

This PR adds mocks for proto interface.
## Which issue(s) does this PR fix/relate to?

- Resolves #2193 

## Have you included tests for your changes?

Yes

## Did you document any new/modified feature?

No

### Notes

dborovcanin reviewed (changes requested):
Don't edit generated files, but use config or CLI commands to generate mock for generated interfaces.

JeffMboya committed:
feature: Add boilerplate file for mockery.yaml

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update mockery.yaml to use correct directory and filename for auth client

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update mockery.yaml to use correct directory and filename for auth client

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update mockery.yaml to use correct interface name

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update boilerplate.txt with SPDX license identifier

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update mockery.yaml  with SPDX license identifier

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

JeffMboya committed:
refactor: add proto interface mocks to CI

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate mocks for proto interface

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update mockery.yaml to use correct directory and filename for auth client

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: delete duplicate files

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: Update boilerplate.txt with copyright information

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate auth_client mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate auth_client mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate auth_client mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate AuthServiceClient mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate AuthServiceClient mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate AuthServiceClient mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update file paths for auth_client mocks

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update bootstrap tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update certs tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update consumers service_test tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update groups tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update invitations tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update consumers api tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update http api tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update journal tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update mqtt tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update sdk tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update readers tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update things tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update twins tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update twins tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update users tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update ws tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: linter

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: linter

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: linter

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing test

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing test

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update Makefile targets

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update Makefile with AuthServiceClient mock config

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update Makefile with AuthServiceClient mock config

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update Makefile with AuthServiceClient mock config

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update boilerplate-file with license info

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: update AuthServiceClient mock configuration and file paths

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: generate mocks for AuthServiceClient

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing tests

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: CI

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: downgrade proto

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

JeffMboya committed:
feat: Update protobuf compiler to v1.34.2 and protoc to v5.27.1

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

JeffMboya committed:
feat: Add PermissionsService/CheckBulkPermissions to replace ExperimentalService/BulkCheckPermission

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

JeffMboya committed:
Fix: Fix infinite loop and improve error handling in Benchmark function

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

feat: add linter timeout

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

feat: add linter timeout

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing test

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing test

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

fix: failing test

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

dborovcanin commented on .github/workflows/tests.yml:
```diff
@@ -27,6 +27,7 @@ jobs:
         uses: golangci/golangci-lint-action@v6
         with:
           version: v1.59.1
+          args: --timeout 3m
```
Since the config is moved from the root, we must pass the config location so golangci-lint runs correctly.

JeffMboya committed:
refactor: add path golangci-lint config file

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>

refactor: add path golangci-lint config file

Signed-off-by: JeffMboya <jangina.mboya@gmail.com>
